README contents (#9)
eileen-kuehn/sorse-data-filter: README.md

# sorse-data-filter

> Intention of this project is to manage visibility of data and automatise several

Reviewer: ```suggestion
This project manages the visibility of data and automatises several
```

workflows centred around the abstract submission process of [SORSE – A Series of
Online Research Softare Events](http://sorse.github.io). It is our aim to ensure

> that no connection between personal data and diversity information based on data
> exported from Indico can be derived.

Reviewer: ```suggestion
that no connection between personal data and diversity information can be derived
from data exported from the [Indico](https://getindico.io) event management platform.
```

The tool currently supports the following automatised workflows:

* ``website``: export of Markdown-formatted data after acceptance to the website

Further workflows are planned including:

* ``mentoring``: export of data to a google drive to support the mentoring process
* ``scheduling``: export of data to a google drive to support the scheduling process
* ``statistics``: export of statistical data to a google drive

# Configuring workflows

The export of data is centred around the concepts of allowing specific data fields
and providing formatting with the help of the templating engine
[Jinja2](https://jinja.palletsprojects.com/en/2.11.x/).

## Configuring allowed fields

Allowed fields within a workflow are specified in the
[workflows.yaml file](workflows.yaml). For each workflow an entry is created that
holds a section for ``allow_lists``. The entries in this field follow the model
specified in [models](models) and include the ``contribution`` itself, ``person``s
and ``questionnaire``s holding specific information for each contribution type
as well as diversity information.

## Configuring templates

Templates that can be used are contained in the [templates folder](templates).

> For example the template for the website workflow is [website.md](templates/website.md).

Reviewer: ```suggestion
For example the template for the website workflow is [templates/website.md](templates/website.md).
```
